### ISSUE TITLE
A tool for doing mind maps

### ISSUE THREAD

**Issue by wel97459** [Frooxius/NeosPublic#220]
Would be cool to have away of doing mind maps.

Basic concept is to have a root node or idea, with nodes branch off each other that describes smaller parts of the bigger consecpt. Would also be cool to have away to link to other types of content not just nodes but images and 3d objects.

Here's a picture:
![apps 38049 9007199266252068 5e2dac67-26d9-453e-806f-6ecbdbb840dd](https://user-images.githubusercontent.com/12990640/54451839-f8c1a580-4710-11e9-95a5-52eacd00d1e2.png)

**Comment by sirkitree** [comment]
Yeah, even a generalized tool that works like the logix tool, and its ability to connect with the wire component. Instead of spawning nodes, you spawn basic geometry or text input nodes, and could then connect them. 🤔 

Been trying to figure out how that works so I can adapt it for a data visualization concept but am having a good time figuring out how that component works. Maybe I'll try adapting the tool instead...

**Comment by Simulacron3** [comment]
Maybe you have something simpler in mind, but the basic way is to have a database behind it, so there is that whole dimension. From there, it would be a matter of visualizing the data with a display of nodes that we can use as an interface to with the database (selecting, displaying linked nodes by relationship & depth, etc.), all nontrivial I would guess. Are there some existing functions or database/visualization framework that could be shoe-horned into Neo to do this sort of thing?

**Comment by sirkitree** [comment]
I came across this project recently which has some unity code (and others) working off of public blockchain transaction records, great use case IMHO: https://blockchain3d.info/ - Not sure if any code could be reused from this, but the visual/concepts I'm into.

**Comment by Simulacron3** [comment]
blockchain3d is generic (not restricted to blockchain visualization) for Unity and is open-source, so promising.

**Comment by sirkitree** [comment]
Been playing around with Noda a little bit. It's a nice implementation of mind maps in VR. I could foresee a tool that works similarly in Neos. It's process is:
1. create a shape with primary action
1. secondary action to edit the shape to add a label, change color, etc.
1. create another shape with primary action
1. primary action on an existing shape and drag to the second shape to create a line connection
1. secondary action on the line to edit the line

I like it's simplicity and might be fairly easy to replicate in Neos.

**Comment by Simulacron3** [comment]
> Been playing around with Noda a little bit. 
What particularly piqued my interest was voice input as alternative to keyboards. Have you tried that?

**Comment by sirkitree** [comment]
> What particularly piqued my interest was voice input as alternative to keyboards. Have you tried that?

I have in a few VR apps now, and it's a much more natural means of input for sure. Most of the time it's never quite accurate though and I have to go back and correct things with a keyboard anyway, but I do think that this is getting better and should be included in Neos at some point. See #85

**Comment by sirkitree** [comment]
There is a public mind map world in which you can use some tools to create these. 

I also worked with some other community members on a tool that is not yet released, but will allow you to go into each mindmap node. Let me know if you'd like a private demo of the application in Neos! :)
